resident-services/README.md

```diff
@@ -0,0 +1,72 @@
+### Contains
+* This folder contains performance test script of Resident Service Endpoints.
+    1. Authorize​ Admin​ Validate Token 
+    2. Login​ Redirect URI 
+    3. VID Policy
+    4. Generate VID
+    5. Get VIDs of the Resident
+    6. Service History
+    7. Revoke VID
+    8. Auth Lock Unlock
+    9. Auth Lock Status 
+    10. Request Card VID 
+    11. Service History Download
+
+
+* Open source Tools used,
+    1. [Apache JMeter](https://jmeter.apache.org/)
+
+### How to run performance scripts using Apache JMeter tool
+* Download Apache JMeter from https://jmeter.apache.org/download_jmeter.cgi
+* Download scripts for the required module.
+* Start JMeter by running the jmeter.bat file for Windows or jmeter file for Unix. 
+* Validate the scripts for one user.
+* Execute a dry run for 10 min.
+* Execute performance run with various loads in order to achieve targeted NFR's.
+
+### Setup points for Execution
+
+* Create Identities in MOSIP Authentication System (Setup) : This thread contains the authorization api's for regproc and idrepo from which the auth token will be generated. There is set of 4 api's generate RID, generate UIN, add identity and add VID. From here we will get the VID's which can be further used as individual id for required Resident Service endpoints. These 4 api's are present in the loop controller where we can define the number of samples for creating identities in which "addIdentitySetup" is used as a variable.
+
+* Resident Id Access Token Creation (Setup) : This thread contains 4 esignet api's Oauth details, Send OTP, Authentication, Authorization Code and 1 reisdent Login​ Redirect URI api.  After the login is successful, the resident will be redirected to the resident portal’s logged-in page. From Login​ Redirect URI api will get the id and access token which will be used further in the headers of resident service api's.
+
+* For execution purpose neeed to check for the mentioned properties: 
+   * esignet default properties: Update the value for the properties according to the execution setup. Perform the execution for esignet api's with redis setup. So check for the redis setup accordingly.
+          mosip.esignet.cache.size - Enabled while not using the redis setup. Can keep the cache size around more than 100k.
+          mosip.esignet.cache.expire-in-seconds (authcodegenerated) - 21600
+          mosip.esignet.access-token-expire-seconds - 86400
+          mosip.esignet.id-token-expire-seconds - 86400
+          spring.cache.type=redis - check for this property and enable the redis.
+   * application default properties: Update the value for the below property.
+          mosip.kernel.otp.expiry-time - 86400
+   * id-authentication default properties: Update the value for the below properties.
+          otp.request.flooding.max-count - 100000
```

Comment:
Give justification for what purpose these properties are changed

Reply:
Done